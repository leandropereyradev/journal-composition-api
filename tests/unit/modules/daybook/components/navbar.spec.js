import { shallowMount } from "@vue/test-utils";

import Navbar from "@/modules/daybook/components/Navbar.vue";
import createVuexStore from "../../../mocks/createVuexStore";
import mockRouterCompositionAPI from "../../../mocks/routerCompositionAPI";

mockRouterCompositionAPI();

describe("Navbar component", () => {
  const store = createVuexStore({
    user: {
      name: "Juan Carlos",
      email: "juan@gmail.com",
    },
    status: "authenticated",
    idToken: "ABC",
    refreshToken: "XYZ",
  });

  beforeEach(() => jest.clearAllMocks());

  test("Should display the component correctly", () => {
    const wrapper = shallowMount(Navbar, {
      global: {
        plugins: [store],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Should log out and redirect when clicking logout", async () => {
    const wrapper = shallowMount(Navbar, {
      global: {
        plugins: [store],
      },
    });

    await wrapper.find("button").trigger("click");

    expect(wrapper.router.push).toHaveBeenCalledWith({ name: "login" });

    expect(store.state.auth).toEqual({
      user: null,
      status: "not-authenticated",
      idToken: null,
      refreshToken: null,
    });
  });
});
