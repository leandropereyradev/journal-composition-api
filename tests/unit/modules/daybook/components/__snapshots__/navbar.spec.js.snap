// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Should display the component correctly 1`] = `
<nav class="navbar bg-primary"><a class="navbar-brand text-white" href=""><img src="" alt="Vue Logo" class="d-inline-block align-text-top mx-2" height="24"> Juan Carlos</a>
  <div class="d-flex"><button class="btn btn-outline-info"><i class="fa fa-sign-out-alt"></i></button></div>
</nav>
`;
